test/Infrastructure.IntegrationTest/AdminConsole/Repositories/OrganizationEnableCollectionEnhancementTests.cs

```diff
@@ -0,0 +1,601 @@
+        var group2 = await CreateGroup(organization, accessAll: true, groupRepository, orgUser);
+
+        var collection1 = await CreateCollection(organization, collectionRepository);
+        var collection2 = await CreateCollection(organization, collectionRepository);
+        var collection3 = await CreateCollection(organization, collectionRepository);
+
+        await organizationRepository.EnableCollectionEnhancements(organization.Id);
+
+        var (updatedOrgUser, collectionAccessSelections) = await organizationUserRepository.GetDetailsByIdWithCollectionsAsync(orgUser.Id);
+
+        Assert.Equal(OrganizationUserType.User, updatedOrgUser.Type);
+
+        // OrgUser has direct Can Manage access to all collections
+        Assert.Equal(3, collectionAccessSelections.Count);
+        Assert.Contains(collectionAccessSelections, cas =>
+            cas.Id == collection1.Id &&
+            cas is { HidePasswords: false, ReadOnly: false, Manage: true });
+        Assert.Contains(collectionAccessSelections, cas =>
+            cas.Id == collection2.Id &&
+            cas is { HidePasswords: false, ReadOnly: false, Manage: true });
+        Assert.Contains(collectionAccessSelections, cas =>
+            cas.Id == collection3.Id &&
+            cas is { HidePasswords: false, ReadOnly: false, Manage: true });
+
+        // Assert: group should only have Can Edit permissions (making sure no side-effects from the Manager migration)
+        var (updatedGroup1, updatedGroupAccess1) = await groupRepository.GetByIdWithCollectionsAsync(group1.Id);
+        Assert.Equal(3, updatedGroupAccess1.Count);
+        Assert.Contains(updatedGroupAccess1, cas =>
+            cas.Id == collection1.Id &&
+            cas is { HidePasswords: false, ReadOnly: false, Manage: false });
+        Assert.Contains(updatedGroupAccess1, cas =>
+            cas.Id == collection2.Id &&
+            cas is { HidePasswords: false, ReadOnly: false, Manage: false });
+        Assert.Contains(updatedGroupAccess1, cas =>
+            cas.Id == collection3.Id &&
+            cas is { HidePasswords: false, ReadOnly: false, Manage: false });
+
+        var (updatedGroup2, updatedGroupAccess2) = await groupRepository.GetByIdWithCollectionsAsync(group2.Id);
+        Assert.Equal(3, updatedGroupAccess2.Count);
+        Assert.Contains(updatedGroupAccess2, cas =>
+            cas.Id == collection1.Id &&
+            cas is { HidePasswords: false, ReadOnly: false, Manage: false });
+        Assert.Contains(updatedGroupAccess2, cas =>
+            cas.Id == collection2.Id &&
+            cas is { HidePasswords: false, ReadOnly: false, Manage: false });
+        Assert.Contains(updatedGroupAccess2, cas =>
+            cas.Id == collection3.Id &&
+            cas is { HidePasswords: false, ReadOnly: false, Manage: false });
+    }
+
+    [DatabaseTheory, MssqlDatabaseData]
+    public async Task Migrate_CustomUser_WithEditAssignedCollections_WithAccessAll_GivesCanManageAccessToAllCollections(
```

Comment:
Note: these custom user tests are identical to the manager tests except for the call to `CreateOrganizationUser` and asserting their role (which unlike the manager role is not changed).

There is maybe a more DRY way of structuring this but I went for the simple copy + paste solution given that these tests will not be around for very long.